src/test/groovy/com/palantir/gradle/ConfigurationCacheTest.groovy

```diff
@@ -0,0 +1,46 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle
+
+import nebula.test.IntegrationTestKitSpec
+
+class ConfigurationCacheTest extends IntegrationTestKitSpec {
+
+    def setup() {
+        // language=Gradle
+        buildFile << """
+            apply plugin: 'com.palantir.git-version'
+            apply plugin: 'java-library'
+
+            repositories {
+                mavenCentral()
+                mavenLocal()
+            }
+
+            version gitVersion()
+        """.stripIndent(true)
+
+        definePluginOutsideOfPluginBlock = true
+        keepFiles = true
+    }
+
+    def "build task runs with configuration cache"() {
+        expect:
+        createRunner('build', '--dry-run', '--configuration-cache')
+                .build()
+    }
```

Comment:
I'm not sure what the best way to test this is as I haven't looked in the details, but it should also be the case that we figure out whether or not hte configuration cache was used a second time round, which is the eventual goal right

Reply:
Good idea! Let's test for this for all the `ConfigurationCacheTest`s we write for the other plugins as well.